components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1943,11 +1906,92 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         )
     }
 
+    /// Check if merge target region in kv engine is fresher than target region.
+    /// It should be called when target region is not in region map in memory.
+    /// If everything is ok, the answer should always be true. So if not, error log will be printed.
+    fn is_local_merge_target_region_fresher(&self, target_region: &metapb::Region) -> Result<bool> {
+        let target_region_id = target_region.get_id();
+        let target_peer_id = util::find_peer(target_region, self.ctx.store_id())
+            .unwrap()
+            .get_id();
+
+        let state_key = keys::region_state_key(target_region_id);
+        if let Some(target_state) = self
+            .ctx
+            .engines
+            .kv
+            .get_msg_cf::<RegionLocalState>(CF_RAFT, &state_key)?
+        {
+            if util::is_epoch_stale(
+                target_region.get_region_epoch(),
+                target_state.get_region().get_region_epoch(),
+            ) {
+                return Ok(true);
+            }
+            // The local target region epoch is staler than target region's.
+            // Check peer id because it may be destroyed by receiving gc msg rather than applying conf change log.
```

Comment:
```suggestion
            // In the case where the peer is destroyed by receiving gc mag rather than applying conf change, the epoch may staler but it's legal, so check peer id to assure that.
```
not quite easy to understand the reason why need to check it, may this is better?

Reply:
Yes. I change it.